src/js/rtc_peer_connection_factory.js

```diff
@@ -66,22 +66,22 @@ export default class RtcPeerConnectionFactory {
             clearTimeout(self._idleRtcPeerConnectionTimerId);
             self._idleRtcPeerConnectionTimerId = null;
         }
-        setTimeout(() => {
-            self._requestPeerConnection();
-        }, 0);
+        self._requestPeerConnection();
         return pc;
     }
 
-
     _requestPeerConnection() {
         var self = this;
-        if (self._browserSupported) {
+        if (!self._peerConnectionRequestInFlight && self._browserSupported) {
+            self._peerConnectionRequestInFlight = true;
             self._requestIceAccess().then(function (response) {
                     self._pc = self._createRtcPeerConnection(response);
+                    self._peerConnectionRequestInFlight = false;
                     self._idleRtcPeerConnectionTimerId = setTimeout(hitch(self, self._refreshRtcPeerConnection), RTC_PEER_CONNECTION_IDLE_TIMEOUT_MS);
```

Comment:
timeoutId is not cleaned up in _refreshRtcPeerConnection. While clearTimeout does not throw, I think it's still better to have a more explicit life cycle there

Reply:
Yes, updated _refreshRtcPeerConnection to set _idleRtcPeerConnectionTimerId to null